pandas/core/internals/blocks.py

```diff
@@ -2114,10 +2114,59 @@ def get_values(self, dtype: DtypeObj | None = None) -> np.ndarray:
     def values_for_json(self) -> np.ndarray:
         return self.values
 
+    @cache_readonly
+    def is_numeric(self) -> bool:
+        dtype = self.values.dtype
+        kind = dtype.kind
 
-class NumericBlock(NumpyBlock):
-    __slots__ = ()
-    is_numeric = True
+        if kind in "fciub":
+            return True
+        else:
+            return False
+
+    @cache_readonly
+    def is_object(self) -> bool:
+        if self.values.dtype.kind == "O":
+            return True
+        else:
+            return False
+
+    @maybe_split
+    def convert(
+        self,
+        *,
+        copy: bool = True,
+        using_cow: bool = False,
+    ) -> list[Block]:
+        """
+        Attempt to coerce any object types to better types. Return a copy
+        of the block (if copy = True).
+        """
+        if not self.is_object:
+            return super().convert(copy=copy, using_cow=using_cow)
```

Comment:
since the base class method is a 1-liner let's just put this whole method there and inline that one line here